types/validator_set.go

```diff
@@ -0,0 +1,16 @@
+package types
+
+import (
+	abci "github.com/tendermint/abci/types"
+)
+
+type Validator = abci.Validator
+
+type ValidatorSetKeeper interface {
+	Validators(Context) []*Validator
+	Size(Context) int
+	IsValidator(Context, Address) bool
+	GetByAddress(Context, Address) (int, *Validator)
+	GetByIndex(Context, int) *Validator
+	TotalPower(Context) Rat
```

Comment:
This is interesting - so an abci.Validator doesn't actually store power as a `Rat` it stores it as an int64 - the stake module however stores each validator's power as a `Rat` and also contains some other values (see: https://github.com/cosmos/cosmos-sdk/blob/develop/x/stake/types.go#L153) not sure if we should have the validator type reflect what's in `stake`..... what are you thoughts?

Reply:
Either we use a new validator type, which is more similar to stake.Validator, or the TotalPower should be an int64. I'm leaning to using `Rat`'s everywhere within the SDK myself - for consistency